core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiServiceSpec.scala

```diff
@@ -1615,6 +1615,65 @@ class EntityApiServiceSpec extends ApiServiceSpec {
         assertResult(StatusCodes.Conflict) {
           status
         }
+
+        assertResult(EntityCopyResponse(Seq.empty, Seq(EntityHardConflict("Sample", "sample1")), Seq.empty)) {
+          responseAs[EntityCopyResponse]
+        }
+      }
+  }
+
+  it should "return 409 for copying entities into a workspace with subtree conflicts, but successfully copy when asked to" in withTestDataApiServices { services =>
+    val sourceWorkspace = WorkspaceName(testData.workspace.namespace, testData.workspace.name)
+    val entityCopyDefinition1 = EntityCopyDefinition(sourceWorkspace, testData.controlledWorkspace.toWorkspaceName, testData.sample1.entityType, Seq(testData.sample1.name))
+    //this will cause a soft conflict because it references sample1
+    val entityCopyDefinition2 = EntityCopyDefinition(sourceWorkspace, testData.controlledWorkspace.toWorkspaceName, testData.sample3.entityType, Seq(testData.sample3.name))
+
+    Post("/workspaces/entities/copy", httpJson(entityCopyDefinition1)) ~>
+      sealRoute(services.entityRoutes) ~>
+      check {
+        assertResult(StatusCodes.Created, response.entity.asString) {
+          status
+        }
+
+        val copyResponse = responseAs[EntityCopyResponse]
+
+        assertSameElements(Seq(testData.sample1, testData.aliquot1).map(_.toReference), copyResponse.entitiesCopied)
+        assertSameElements(Seq.empty, copyResponse.hardConflicts)
+        assertSameElements(Seq.empty, copyResponse.softConflicts)
+      }
+
+    //test the default case of no parameter set
+    Post("/workspaces/entities/copy", httpJson(entityCopyDefinition2)) ~>
+      sealRoute(services.entityRoutes) ~>
+      check {
+        assertResult(StatusCodes.Conflict, response.entity.asString) {
+          status
+        }
+        assertResult(EntityCopyResponse(Seq.empty, Seq.empty, Seq(EntitySoftConflict(testData.sample3.entityType, testData.sample3.name, Seq(EntitySoftConflict(testData.sample1.entityType, testData.sample1.name, Seq.empty), EntitySoftConflict(testData.sample1.entityType, testData.sample1.name, Seq(EntitySoftConflict(testData.aliquot1.entityType, testData.aliquot1.name, Seq.empty)))))))) {
```

Comment:
Please format this line (and 1663 - hm, maybe make this a val?) in a way that shows the hierarchy.  Something like:

```
assertResult(EntityCopyResponse(Seq.empty, Seq.empty, Seq(
          EntitySoftConflict(testData.sample3.entityType, testData.sample3.name, Seq(
            EntitySoftConflict(testData.sample1.entityType, testData.sample1.name, Seq.empty),
            EntitySoftConflict(testData.sample1.entityType, testData.sample1.name, Seq(EntitySoftConflict(testData.aliquot1.entityType, testData.aliquot1.name, Seq.empty)))))))) {
```